ui/src/components/ChatComponent/chatMessages.js

```
@@ -0,0 +1,29 @@
+function normalizeDateToISO(dateString) {
+  let normalizedDate = new Date().toISOString();
+  if (dateString) {
+    try {
+      const parsedDate = new Date(dateString);
+      if (!Number.isNaN(parsedDate.getTime())) {
+        normalizedDate = parsedDate.toISOString();
+      }
+    } catch {
+      // Ignore
+    }
+  }
+  return normalizedDate;
+}
+
+export function processChatMessageRecord(record, currentUsername) {
+  const isSelf = record.username === currentUsername;
+  const normalizedDate = normalizeDateToISO(record.date);
```

I don't think you need that `normalizeDateToISO` function. I would also throw an error if the date is invalid (instead of ignoring the problem), as it could help uncover a bug.

```js
const date = record.date ? new Date(record.date) : new Date();
if (Number.isNaN(date.getTime()) {
  throw new Error('Invalid date');
}

// ...
date: date.toISOString(),
```